protocol-models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -315,6 +315,30 @@ definitions:
       - RUNNING # Stream has read its first byte/message
       - COMPLETE # Stream has completed executing without interruption or error
       - INCOMPLETE # Stream has stopped due to an interruption or error
+  AirbyteStreamStatusReasonType:
+    type: string
+    description: >
+      Type of reason
+    enum:
+      - RATE_LIMITED
+  AirbyteStreamStatusRateLimitedReason:
+    type: object
+    description: Rate Limited Information
+    properties:
+      quota_reset:
+        description: "Optional time in ms representing when the API quota is going to be reset"
+        type: number
```

Comment:
any reason this isn't just a big int? do we need decimals?

Reply:
@cgardens I am just following the convention. We are using number for timestamp attributes. For instance take a look at the emitted_at in `AirbyteTraceMessage`, its defined as a number 
```
      emitted_at:
        description: "the time in ms that the message was emitted"
        type: number
```

Reply:
@cgardens @subodh1810 This looks like a bug in the `AirbyteTraceMessage`.  It was not obvious to me that `integer` was not an integer data type and is actually a big integer or that `number` is a floating point/double.  Should we attempt to fix the `AirbyteTraceMessage` either as part of this PR or in a separate PR?

Reply:
* let's do it correctly here and use integer. 
* let's create an issue to fix airbyte trace meesage and add a todo in the code that that should have been number. don't think this is a good moment to try to fix this old mistake. just want to do it right going forward.

Reply:
@cgardens @jdpgrailsdev added the TODOs in the protocol yaml and created a github issue 
https://github.com/airbytehq/airbyte/issues/38163